### PR TITLE
[SPARK-5399] tree Losses strings should match loss names

### PR DESCRIPTION
**Pull request by Lewuathe**
For avoiding confusion about loss function in `mllib.tree.loss`, rename legal arguments. 
Though it does not seem any documentation about these loss function for me, I want to update that if there is appropriate section.

### CODE REVIEW

File: mllib/src/test/scala/org/apache/spark/mllib/tree/LossesSuite.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.apache.spark.mllib.tree
```
Comment (srowen) [inline mllib/src/test/scala/org/apache/spark/mllib/tree/LossesSuite.scala]: (Copyright header here). Looks reasonable but I don't know the reason for the difference in the first place. This `Losses` class doesn't seem to be used anyway? I can't find it at least.

Comment (jkbradley) [inline mllib/src/test/scala/org/apache/spark/mllib/tree/LossesSuite.scala]: Huh, good point.  It must have been left over after some other change.  Perhaps we should just remove the whole class.

**Comment by Lewuathe** [comment]
`Losses` is added in this commit. In that time, this is used by `GradientBoosting`.
https://github.com/apache/spark/commit/8602195510f5821b37746bb7fa24902f43a1bd93

But renamed with [this commit](https://github.com/apache/spark/commit/15cacc81240eed8834b4730c5c6dc3238f003465). These are not seemed to be used any more. Should I remove whole `Losses` class?

**Comment by jkbradley** [comment]
Right, that was where we removed the train(long list of parameters) methods.  It's odd to keep it if we do not use it anywhere.  On the other hand, I like the idea of enforcing standardized String names for Losses.  What about adding a loss option to the GradientBoostedTreesRunner example?  Does that seem useful?

**Comment by srowen** [comment]
At this point I'd say let's just remove the class if it's not used.

**Comment by jkbradley** [comment]
@Lewuathe  Would you have time to close this and do a new PR to remove it?  If not, I could make it part of my PR for [https://issues.apache.org/jira/browse/SPARK-6113]

**Comment by Lewuathe** [comment]
@jkbradley Sorry for being late response. I cannot have enough time to submit this patch, so it`s OK to make this part of your PR. Thank you!

**Comment by jkbradley** [comment]
Sure, I'll do that.  Would you mind closing this PR then?

**Comment by Lewuathe** [comment]
@jkbradley Yes. Thank you.

**Comment by srowen** [comment]
@Lewuathe you will have to close this PR yourself.

**Comment by Lewuathe** [comment]
I'm sorry. I close this. Thank you.

**Pull request closed without merging**